### PR TITLE
 Add Dockerfile to make manylinux1 wheels (e.g. for PyPI)

### PR DESCRIPTION
**Pull request by ihnorton**
Adds Dockerfile and associated build script to create manylinux1
wheels for TileDB-Py targeting py27 and py37
- introduce Dockerfile inheriting from manylinux1
- improve build system support for linux wheel packaging

(also includes a fix to allow overriding the cmake binary, which should help to fix doc builds on RTD)

### CODE REVIEW

**Comment by ihnorton** [comment]
Mac and windows failures are unrelated -- in core library build (mac is fixed in https://github.com/TileDB-Inc/TileDB/pull/1196; will fix windows).

File: misc/pypi_linux/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+#    2) perl-5.10.0, buit above, is required to build OpenSSL
+RUN cd /home/tiledb/ && \
+  source $HOME/perl5/perlbrew/etc/bashrc && \
+  perlbrew use perl-5.10.0 && \
+  git clone https://github.com/TileDB-Inc/TileDB && \
+  git -C TileDB checkout $TILEDB_VERSION && \
+  mkdir build && \
+  cd build && \
+  $CMAKE -DTILEDB_S3=ON -DTILEDB_HDFS=ON -DTILEDB_TESTS=OFF \
+         -DTILEDB_FORCE_ALL_DEPS:BOOL=ON -DSANITIZER="OFF;-DCOMPILER_SUPPORTS_AVX2:BOOL=FALSE" \
```
Comment (Shelnutt2) [inline misc/pypi_linux/Dockerfile]: The AVX2 check doesn't disable AVX (the first version of avx). If we are trying to target all x86_64 machines then we probably need to set the march to either the generic x86_64 or some arch that does not have AVX. There is a performance penalty for disabling AVX completely though, we might want to benchmark the performance hit before we push a completely generic lib.

File: misc/pypi_linux/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+#    1) we are using auditwheel from https://github.com/pypa/auditwheel
+#       this verifies and tags wheel products with the manylinux1 label,
+#       and allows us to build libtiledb once, install it to a normal
+#       system path, and then use it to build wheels for all of the python
+#       versions.
+#    2) perl-5.10.0, buit above, is required to build OpenSSL
+RUN cd /home/tiledb/ && \
+  source $HOME/perl5/perlbrew/etc/bashrc && \
+  perlbrew use perl-5.10.0 && \
+  git clone https://github.com/TileDB-Inc/TileDB && \
```
[the remark this answers is not in the document]
Comment (ihnorton) [reply misc/pypi_linux/Dockerfile]: The next line checks out a branch (TILEDB_VERSION)

[the remark this answers is not in the document]
Comment (Shelnutt2) [reply misc/pypi_linux/Dockerfile]: Doh! Missed that, thanks for pointing it out.

File: misc/pypi_linux/Dockerfile (hunk 1 above)
Comment (ihnorton) [reply misc/pypi_linux/Dockerfile]: As far as I can tell, we don't explicitly add `-mavx`, so removing `-mavx2` means we will just use the GCC defaults (`-march=x86-64`) unless something else is passed in CFLAGS.

**Pull request merged**